scenes/hud.gd

```diff
@@ -0,0 +1,36 @@
+extends CanvasLayer
+
+signal start_game
+
+var isPaused = false
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	# Need to be listening always for input so can use button to open/close Menu
+	process_mode = Node.PROCESS_MODE_ALWAYS
+
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+func _process(delta):
+	if(isPaused):
+		$Menu/GameStartButton.text="Restart"
+	else:
+		$Menu/GameStartButton.text="Start Game"
+	if Input.is_action_just_released("menu"):
+		if(isPaused):
+			$Menu.hide();
+			isPaused = false
+			get_tree().paused = false
+		else:
+			$Menu.show();
+			isPaused = true
+			get_tree().paused = true
+
+
+func _on_quit_button_pressed():
+	get_tree().quit()
+
+func _on_gamestart_button_pressed():
+	if(isPaused):
+		isPaused = false
+		get_tree().paused=false
+	start_game.emit() 
```

Comment:
I think we should also have a function to update the HealthHearts based on the health property of the player. Maybe not in this PR, but once we make the player script. Since the player script should emit a LoseHealth signal, and the HUD update should be in the HUD script for clarity of code (all hud changes is in hud script)

Reply:
Agree, my plan was to update once we have an updated player script/enemies to make a follow up pr/story for handling user/enemy health visulization